python/tvm/relay/to_onnx.py

```diff
@@ -0,0 +1,422 @@
+        if num_newaxis:
+            output_name = 'inter{}'.format(node['output_names'][0])
+            transpose_node = onnx.helper.make_node('Unsqueeze',
+                                                   [node['input_names'][1]],
+                                                   [output_name],
+                                                   **{'axes': tuple(range(1, num_newaxis+1))})
+            model_container.add_nodes([transpose_node])
+        else:
+            output_name = node['input_names'][1]
+
+        inputs = [node['input_names'][0], output_name]
+        matmul_node = onnx.helper.make_node('Add', inputs, node['output_names'])
+        model_container.add_nodes([matmul_node])
+
+
+relay_to_onnx_op_mapping = {
+    'reshape': Reshape,
+    'conv2d': Conv,
+    'add': rename('Add'),
+    'relu': rename('Relu'),
+    'transpose': Transpose,
+    'dense': MatMul,
+    'max_pool2d': MaxPool,
+    'batch_flatten': Flatten,
+    'multiply': rename('Mul'),
+    'bias_add': BiasAdd,
+    'batch_norm': BatchNormalization,
+    'global_avg_pool2d': rename('GlobalAveragePool'),
+    'concatenate': Concat,
+    'dropout': Dropout,
+    'avg_pool2d': AveragePool,
+    'divide': rename('Div')
+}
+
+
+class ModelContainer(object):
+    """ A container class to hold  different attributes of ONNX model graph
+    """
+
+    def __init__(self, name):
+        self._name = name
+        self._inputs = []
+        self._outputs = []
+        self._nodes = []
+        self._initializers = []
+
+    def add_inputs(self, inputs):
+        self._inputs.extend(inputs)
+
+    def add_outputs(self, outputs):
+        self._outputs.extend(outputs)
+
+    def add_nodes(self, nodes):
+        self._nodes.extend(nodes)
+
+    def add_initializers(self, initializers):
+        self._initializers.extend(initializers)
+
+    def make_model(self):
+        graph_def = onnx.helper.make_graph(
+            self._nodes,
+            self._name,
+            self._inputs,
+            self._outputs,
+            self._initializers
+        )
+
+        return onnx.helper.make_model(graph_def, producer_name='relay')
+
+
+class RelayToONNXConverter(object):
+    """A helper class converting topologically sorted  Relay Node list to ONNX model
+
+    Parameters
+    ----------
+    name : str
+       name of the model
+
+    node_list : list
+        topologically sorted Relay Node entry list
+    """
+
+    def __init__(self, name, node_list, params):
+        self._name = {}
+        self._mc = ModelContainer(name)
+        self._node_list = node_list
+        self._params = params
+
+    def convert_to_onnx(self):
+        """ Loop through topologically sorted list of Relay nodes and generate a ONNX model"""
+        for idx, node_entry in enumerate(self._node_list):
+            out_idx = idx
+            node = node_entry['node']
+            if isinstance(node, Call):
+                self._add_node(node_entry, idx)
+            elif isinstance(node, Var):
```

Comment:
This may be incorrect. Relay has Module that contains functions. Each function can have args that are vars, but not all these vars are model inputs. You probably need to take the args of the entry function only, e.g. main.

Reply:
This function loops through "main" function only. Please refer https://github.com/neo-ai/tvm/pull/64/files#diff-748c31d6923a2e93939417b2d077734bR403. The support for other functions is not added yet.